MuonAnalysis/MuonAssociators/interface/L1MuonMatcherAlgo.h

```diff
@@ -133,16 +173,21 @@ class L1MuonMatcherAlgo {
         template<typename Collection, typename Selector>
         int matchGeneric(TrajectoryStateOnSurface &propagated, const Collection &l1, const Selector &sel, float &deltaR, float &deltaPhi) const ;
 
-
         /// Add this offset to the L1 phi before doing the match, to correct for different scales in L1 vs offline
         void setL1PhiOffset(double l1PhiOffset) { l1PhiOffset_ = l1PhiOffset; }
 
     private:
         PropagateToMuon prop_;
 
+	bool useStage2L1_;
+
         typedef StringCutObjectSelector<l1extra::L1MuonParticle> L1Selector;
         /// Preselection cut to apply to L1 candidates before matching
-        L1Selector preselectionCut_;
+        L1Selector preselectionCutL1_;
+
+        typedef StringCutObjectSelector<l1t::Muon> L1TSelector;
+        /// Preselection cut to apply to L1 candidates before matching
+        L1TSelector preselectionCutL1T_;
```

Comment:
Se ci metti il "true" come secondo parametro del template (o del
costruttore), lui vede tutti i metodi anche delle classi derivate.

On Fri, May 13, 2016 at 12:33 PM, Carlo Battilana <notifications@github.com>
wrote:

> In MuonAnalysis/MuonAssociators/interface/L1MuonMatcherAlgo.h
> <https://github.com/gpetruc/cmssw/pull/8#discussion_r63163535>:
>
> >          typedef StringCutObjectSelector<l1extra::L1MuonParticle> L1Selector;
> >          /// Preselection cut to apply to L1 candidates before matching
> > -        L1Selector preselectionCut_;
> > +        L1Selector preselectionCutL1_;
> > +
> > +        typedef StringCutObjectSelector<l1t::Muon> L1TSelector;
> > +        /// Preselection cut to apply to L1 candidates before matching
> > +        L1TSelector preselectionCutL1T_;
>
> Ci provo ma ho una domanda stupida, se faccio come suggerisci poi il
> selettore accede correttamente ai membri che ci sono solo nelle classi
> derivate tipo "bx" o vede solo l'interfaccia di reco::Candidate ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/gpetruc/cmssw/pull/8/files/1e8d982ddd89bac6791aaaf20485ada2a3d0b675#r63163535>
>